pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -3476,12 +3476,14 @@ private synchronized void checkReplicatedSubscriptionControllerState(boolean sho
         boolean isCurrentlyEnabled = replicatedSubscriptionsController.isPresent();
         boolean isEnableReplicatedSubscriptions =
                 brokerService.pulsar().getConfiguration().isEnableReplicatedSubscriptions();
+        boolean isReplicationDisEnable = this.topicPolicies.getReplicationClusters().get().size() == 1;
 
-        if (shouldBeEnabled && !isCurrentlyEnabled && isEnableReplicatedSubscriptions) {
+        if (shouldBeEnabled && !isCurrentlyEnabled && isEnableReplicatedSubscriptions && !isReplicationDisEnable) {
             log.info("[{}] Enabling replicated subscriptions controller", topic);
             replicatedSubscriptionsController = Optional.of(new ReplicatedSubscriptionsController(this,
                     brokerService.pulsar().getConfiguration().getClusterName()));
-        } else if (isCurrentlyEnabled && !shouldBeEnabled || !isEnableReplicatedSubscriptions) {
+        } else if (isCurrentlyEnabled && !shouldBeEnabled || !isEnableReplicatedSubscriptions
+                || isReplicationDisEnable) {
             log.info("[{}] Disabled replicated subscriptions controller", topic);
             replicatedSubscriptionsController.ifPresent(ReplicatedSubscriptionsController::close);
```

Comment:
At this line, you removed an exist `replicatedSubscriptionsController` instance, right?

If yes, you should stop the timer task `startNewSnapshot` of `replicatedSubscriptionsController` here

And please add a test for this case.

Reply:
>If yes, you should stop the timer task startNewSnapshot of replicatedSubscriptionsController here

The logic of `ReplicatedSubscriptionsController::close` is as follow:
```
    @Override
    public void close() {
        timer.cancel(true);
    }
```